WireAPI/Sources/WireAPI/APIs/VersionedAPI.swift

```diff
@@ -0,0 +1,38 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+protocol VersionedAPI {
+
+    var apiVersion: APIVersion { get }
+
+}
+
+extension VersionedAPI {
+
+    var pathPrefix: String {
+        switch apiVersion {
+        case .v0:
+            ""
+        default:
+            "/v\(apiVersion.rawValue)"
+        }
+    }
```

Comment:
Is this a pattern for the whole API or only some endpoints?

Reply:
We could also consider configure it on the http client.

Reply:
either way we solve this: protocol or base class I am up for having a solution in order not to specify the version on each `XYZAPIV_` classes

Reply:
Sounds good this way, thanks @typfel